mozilla-release/toolkit/modules/PrivateBrowsingUtils.jsm

```diff
@@ -34,6 +34,8 @@ this.PrivateBrowsingUtils = {
   },
 
   isBrowserPrivate: function(aBrowser) {
+    return aBrowser.loadContext.usePrivateBrowsing;
```

Comment:
Should improve behaviour in other scenarios (like saving files) and respect single tab privateness.